java/http/vertx/java-http-vertx-consumer/src/main/java/HttpKafkaConsumer.java

```diff
@@ -68,190 +69,192 @@ public void start(Future<Void> startFuture) throws Exception {
-        return fut;
+             .putHeader(HttpHeaderNames.CONTENT_LENGTH.toString(), String.valueOf(json.toBuffer().length()))
+             .putHeader(HttpHeaderNames.CONTENT_TYPE.toString(), "application/vnd.kafka.v2+json")
+             .as(BodyCodec.jsonObject())
+             .sendJsonObject(json)
+             .onSuccess(response -> {
+                 if (response.statusCode() == HttpResponseStatus.OK.code()) {
+                     JsonObject body = response.body();
+                     this.consumer = new CreatedConsumer(body.getString("instance_id"), body.getString("base_uri"));
+                     log.info("Consumer created as {}", this.consumer);
+                     promise.complete(consumer);
+                 } else {
+                     promise.fail(new RuntimeException("Got HTTP status code " + response.statusCode()));
+                 }
+             })
+             .onFailure(cause -> {
+                 promise.fail(cause);
+             });
+        return promise.future();
     }
 
     private Future<List<ConsumerRecord>> poll() {
-        Future<List<ConsumerRecord>> fut = Future.future();        
+        Promise<List<ConsumerRecord>> promise = Promise.promise();
 
         log.info("Poll ...");
         this.client.get(this.consumer.getBaseUri() + "/records?timeout=" + this.config.getPollTimeout())
-            .putHeader(HttpHeaderNames.ACCEPT.toString(), "application/vnd.kafka.json.v2+json")
-            .as(BodyCodec.jsonArray())
-            .send(ar -> {
-                if (ar.succeeded()) {
-                    HttpResponse<JsonArray> response = ar.result();
-                    if (response.statusCode() == HttpResponseStatus.OK.code()) {
+                .putHeader(HttpHeaderNames.ACCEPT.toString(), "application/vnd.kafka.json.v2+json")
+                .as(BodyCodec.jsonArray())
+                .send(ar -> {
+                    if (ar.succeeded()) {
+                        HttpResponse<JsonArray> response = ar.result();
+                        if (response.statusCode() == HttpResponseStatus.OK.code()) {
 
-                        Tracer tracer = GlobalTracer.get();
+                            Tracer tracer = GlobalTracer.get();
 
-                        MultiMap rawHeaders = response.headers();
+                            MultiMap rawHeaders = response.headers();
                         final Map<String, String> headers = new HashMap<>();
                         for (Map.Entry<String, String> header : rawHeaders) {
                             headers.put(header.getKey(), header.getValue());
                         }
 
-                        String operation = "poll";
-                        Tracer.SpanBuilder spanBuilder;
-                        try {
-                            SpanContext parentSpan = tracer.extract(Format.Builtin.HTTP_HEADERS, new TextMapAdapter(headers));
-                            if (parentSpan == null) {
+                            String operation = "poll";
```

Comment:
This seems to be misaligned -> the indentation of line 174 and 180 is different from 175-179.

Reply:
I've realigned the whole HttpKafkaConsumer class to where it was in the previous commits. Hopefully this is correct?